gadopt/stokes_integrators.py

```diff
@@ -281,9 +267,34 @@ def __init__(
                     weak_bc[bc_type] = value
             self.weak_bcs[id] = weak_bc
 
+        # eta is a list of 0, 1 or multiple free surface fields
+        u, p, *eta = fd.split(self.solution)
+        rho_mass = approximation.rho_continuity()
+        stress = approximation.stress(u)
+        eqs_attrs = [
+            {"u": u, "p": p, "T": T, "stress": stress},
```

Comment:
Does u need to be provided to momentum now as it's the trial function?

Reply:
Good catch, thank you.